ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -11628,6 +11628,11 @@ private void setupStats(TableScanDesc tsDesc, QBParseInfo qbp, Table tab, String
   }
 
   private Operator genPlan(QB parent, QBExpr qbexpr) throws SemanticException {
+    if (qbexpr.getOpcode() == QBExpr.Opcode.EXCEPT || qbexpr.getOpcode() == QBExpr.Opcode.EXCEPTALL
+        || qbexpr.getOpcode() == QBExpr.Opcode.INTERSECT || qbexpr.getOpcode() == QBExpr.Opcode.INTERSECTALL) {
+      throw new SemanticException(
+          "Except and intersect operations are only supported with Cost Based Optimizations enabled. Turn on hive.cbo.enable");
```

Comment:
Could we say instead or something like it?
```
"EXCEPT and INTERSECT operations are only supported with Cost Based Optimizations enabled. Please set 'hive.cbo.enable' to true"
```

Reply:
Fixed